searx/resources.py

```diff
@@ -0,0 +1,886 @@
+        _templates = Templates(builtin_path, default_theme)
+    return _templates
+
+class Templates:
+
+    """Managing *templates* aka *themes*.  A *template* consist of a (jinja)
+    template to render HTML on server site and a *theme* which delivers the
+    static files needed by the client site.
+    """
+
+    def __init__(self, builtin_path, default_theme):
+
+        self.themes = dict()
+        self.default_theme = default_theme
+        self.builtin_path = builtin_path
+
+        # register builtin themes
+        for theme_name in self.builtin_path.listdir():
+            template_folder = self.builtin_path / theme_name
+            if theme_name != '__common__' and template_folder.ISDIR:
+                static_url = File('themes') / theme_name
+                theme_folder = SEARX_DIR / 'static' / 'themes' / theme_name
+                self.themes[theme_name] = Theme(
+                    theme_name, theme_folder, static_url, template_folder
+                )
+
+    def get_result_template(self, theme_name, template_name):
+        """Returns the *result* template ``template_name``.  It is used in (jinja)
+        templates to include a template file conditionally.  The returned path
+        of the template file is relative to `Flask.template_folder`_.
+
+        .. _Flask.template_folder:
+            https://flask.palletsprojects.com/en/1.1.x/api/#flask.Flask.template_folder
+        """
+        theme = self.themes[theme_name]
+
+        ret_val = File('result_templates') / template_name
+        fname = theme.result_templates / template_name
+        if fname.EXISTS:
+            ret_val = fname.relpath(theme.template_folder.DIRNAME)
+
+        return ret_val
+
+    def get_favicons(self, theme_name):
+        """Returns a list of favicons, its used in (jinja) templates to add favicons to
+        the *results* (compare :origin:`static/themes/legacy/img/icons`)::
+
+            {% if "icon_"~result.engine~".ico" in favicons %}
+        """
+        theme = self.themes[theme_name]
+        return theme.favicons
+
+class File(str):  # pylint: disable=too-many-public-methods
+    """A path name to a file or folder.  Handling path names more comfortable, e.g.:
+
+    - concatenate path names with the division operator ``/``
+    - call functions like *makedirs* (aka ``mkdir -p``) directly on the path
+      name
+    - get properties like *EXISTS*
+
+    .. code-block:: python
+
+      >>> from searx.resources import File
+      >>> tmp = File('~/tmp') / 'foo'
+      >>> tmp
+      '/home/user/tmp/foo'
+      >>> tmp.EXISTS
+      False
+
+    No additional import, no juggling with ``os.join(...)``, simply slash ``/``
+    and ``foo.<method>`` calls
+
+    .. code-block:: python
+
+      >>> [(tmp/x).makedirs() for x in ('foo', 'bar')]
+      True, True
+      >>> for n in tmp.listdir():
+      ...     print(tmp / n)
+      ...
+      /home/user/tmp/foo
+      /home/user/tmp/bar
+
+    """
+
+    class OS:
```

Comment:
class File helps me to write much more readable code. This implementation here is the surrogate from a [fspath package I maintain](https://return42.github.io/fspath/slides/index.html#/).  The linked slide show from this project might give a impression where it can help to be more expressive and readable.

Reply:
My comment was about the `OS` class which seems to be only an alias to functions can be found in the standard libs of python.

Reply:
> My comment was about the OS class 

Ah, sorry ,,

> which seems to be only an alias to functions can be found in the standard libs of python.

Yes, these are aliases glued at the File object, making imports from os.path obsolete / same topic: [They exists for programmers convenience](https://github.com/asciimoo/searx/pull/1938#discussion_r447614158). Here is a hypothetical  example:
````python
if myfile.DIRNAME == myfile.OS.curdir():
    # file is in the current working directory
    ....
````

If the `File.OS` namespace is a KO criteria [I can clean up the code](https://github.com/asciimoo/searx/pull/1938#discussion_r447614158). But should we really remove all _convenience_ from .. I have a strange feeling with.